src/Test.FunctionalTests.Sarif/TestData/Multitool/ValidateCommand/ExpectedOutputs/SARIF1004.ExpressUriBaseIdsCorrectly_Invalid.sarif

```diff
@@ -124,6 +130,136 @@
             }
           ]
         },
+        {
+          "ruleId": "SARIF1004",
+          "ruleIndex": 0,
+          "level": "error",
+          "message": {
+            "id": "Error_UriBaseIdValueMustNotContainDotDotSegment",
+            "arguments": [
+              "runs[0].originalUriBaseIds.BAD_URI_CONTAINING_DOTDOT",
+              "file:///C:/rules/dir1/dir2/../",
+              "BAD_URI_CONTAINING_DOTDOT"
+            ]
+          },
+          "locations": [
+            {
+              "physicalLocation": {
+                "artifactLocation": {
+                  "index": 0
+                },
+                "region": {
+                  "startLine": 114,
+                  "startColumn": 38
+                }
+              }
+            }
+          ]
+        },
+        {
+          "ruleId": "SARIF1004",
+          "ruleIndex": 0,
+          "level": "error",
+          "message": {
+            "id": "Error_UriBaseIdValueMustEndWithSlash",
+            "arguments": [
+              "runs[0].originalUriBaseIds.BAD_URI_CONTAINING_QUERY",
+              "http://www.contoso.com/catalog/shownew.htm?date=today",
+              "BAD_URI_CONTAINING_QUERY"
+            ]
+          },
+          "locations": [
+            {
+              "physicalLocation": {
+                "artifactLocation": {
+                  "index": 0
+                },
+                "region": {
+                  "startLine": 120,
+                  "startColumn": 37
+                }
+              }
+            }
+          ]
+        },
+        {
+          "ruleId": "SARIF1004",
```

Comment:
>SARIF1004 [](start = 21, length = 9)

Why do the "query" and "fragment" errors show up twice each?

Reply:
because they also do not end with slash. the other violation is for UriBaseIdValueMustEndWithSlash

---
In reply to: [445147588](https://github.com/microsoft/sarif-sdk/pull/1930#discussion_r445147588) [](ancestors = 445147588)

Reply:
Oh :)

---
In reply to: [445150019](https://github.com/microsoft/sarif-sdk/pull/1930#discussion_r445150019) [](ancestors = 445150019,445147588)